pkg/cmd/admin/prune/deployments.go

```diff
@@ -89,11 +90,20 @@ func NewCmdPruneDeployments(f *clientcmd.Factory, parentName, name string, out i
 			case true:
 				deploymentPruneFunc = func(deployment *kapi.ReplicationController) error {
 					describingPruneDeploymentFunc(deployment)
-					err := kclient.ReplicationControllers(deployment.Namespace).Delete(deployment.Name)
-					if err != nil {
-						return err
+					// If the deployment is failed we need to remove its deployer pods, too.
+					if deployutil.DeploymentStatusFor(deployment) == deployapi.DeploymentStatusFailed {
+						dpSelector := deployutil.DeployerPodSelector(deployment.Name)
+						deployers, err := kclient.Pods(deployment.Namespace).List(dpSelector, fields.Everything())
+						if err != nil {
+							glog.V(3).Infof("Cannot list deployer pods for %q: %v", deployment.Name, err)
+						}
+						for _, pod := range deployers.Items {
+							if err := kclient.Pods(pod.Namespace).Delete(pod.Name, nil); err != nil {
+								glog.V(3).Infof("Cannot remove deployer pod %q: %v", pod.Name, err)
```

Comment:
The image pruner seems to print directly to stderr:

https://github.com/openshift/origin/blob/master/pkg/cmd/admin/prune/images.go#L317

I'm not sure which is preferred in this context, just wanted to point out that there's a difference in approaches.